### PR TITLE
Fix for #1806 - interpret ExpiresOn from broker correctly

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/TokenCache.cs [hunk 1]
```diff
@@ -31,6 +31,7 @@ public sealed partial class TokenCache : ITokenCacheInternal
     {
         internal const string NullPreferredUsernameDisplayLabel = "Missing from the token response";
         private const int DefaultExpirationBufferInMinutes = 5;
+        internal const int ExpirationTooLargeInDays = 10 * 365;
```
Comment (jennyf19) [inline src/client/Microsoft.Identity.Client/TokenCache.cs:34]: >ExpirationTooLargeInDays [](start = 27, length = 24)

nit: is the expiration too large or too long? is the value here too big or the date too far off into the future?

File: src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs [hunk 2]
```diff
@@ -142,5 +142,52 @@ internal static MsalTokenResponse CreateFromBrokerResponse(Dictionary<string, st
...
+                };
+            }
+
+            MsalTokenResponse msalTokenResponse = new MsalTokenResponse()
+            {
+                Authority = authResult[BrokerResponseConst.Authority],
+                AccessToken = authResult[BrokerResponseConst.AccessToken],
+                IdToken = authResult[BrokerResponseConst.IdToken],
+                CorrelationId = correlationId, // Android response does not expose Correlation ID
+                Scope = authResult[BrokerResponseConst.AndroidScopes], // sadly for iOS this is "scope" and for Android "scope"
```
Comment (jennyf19) [inline src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs:172]: what's the difference between iOS and Android on the scope value?

File: tests/Microsoft.Identity.Test.Unit.net45/CoreTests/OAuth2Tests/TokenResponseTests.cs [hunk 3]
```diff
@@ -47,5 +51,56 @@ public void JsonDeserializationTest()
                 Assert.IsNotNull(response);
             }
         }
+
+        [TestMethod]
+        public void BrokerTokenResponse()
+        {
+            //            string s2 = @"{""access_token"":""atsecret"",
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Test.Unit.net45/CoreTests/OAuth2Tests/TokenResponseTests.cs:58]: nit: do you need this still?

File: src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs (hunk 2 above)
Comment (trwalke) [inline src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs:172]: did you mean scope vs scopes?

File: tests/Microsoft.Identity.Test.Unit.net45/CoreTests/OAuth2Tests/TokenResponseTests.cs [hunk 4]
```diff
@@ -47,5 +51,56 @@ public void JsonDeserializationTest()
...
+
+            string unixTimestamp = CoreHelpers.DateTimeToUnixTimestamp(DateTimeOffset.UtcNow + TimeSpan.FromMinutes(40));
+
+            string androidBrokerResponse = @"
```
Comment (trwalke) [inline tests/Microsoft.Identity.Test.Unit.net45/CoreTests/OAuth2Tests/TokenResponseTests.cs:63]: nit: could move to test constants

File: src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs (hunk 2 above)
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs:172]: Sorry, bad comment. It's supposed to say "scope" vs "scopes"

Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs:172]: yes

File: src/client/Microsoft.Identity.Client/TokenCache.cs (hunk 1 above)
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/TokenCache.cs:34]: Too long, too long... :)

**Pull request merged**